requests/cookies.py

```diff
@@ -144,11 +144,10 @@ def remove_cookie_by_name(cookiejar, name, domain=None, path=None):
     """
     clearables = []
     for cookie in cookiejar:
-        if cookie.name != name:
-            continue
-        if domain is not None and domain != cookie.domain:
-            continue
-        if path is not None and path != cookie.path:
+        bool_cookie = (cookie.name != name)
```

Comment:
Your first link has nothing to do with this pattern. Your first link objects to having a *nested* if conditional that represents the entire body of the outer *if* conditional. This is not that. This has consecutive if conditionals *at the same level* that each continue the loop. Sandi *also* complains about nested conditionals, which again is not something we have here. Sandi's complaints are about differing levels of abstractions, which are not true here. We are operating at one

This is widely considered a *very good* pattern for handling complex conditional logic in loops because it splits each check out to its own line that can be considered by itself.

And, to be clear, the variable names you're using here actively reduce the clarity of this block. `bool_cookie` is a very problematic variable name because it makes the conditional at the bottom really unclear. *All* of these boolean metavariables are to do with cookies! The question is, is this cookie the one we're looking for.

There is a problem here with assuming that because there are multiple `if` conditionals in this code it is unclear. This code is about as clear as it gets. Each `if` conditional follows the exact same pattern except for the first, which does not allow `name` to be `None`. All the rest say "if this parameter was passed to us, check it matches the cookie we're looking at". There is nothing complex or complicated about this code.

The *best* way to clean this code up would be to add a comment simply explaining that. It'd be maybe two lines:

```python
# Loop over the cookies in the jar, to find any that match the filter
# criteria passed by the user (same name, domain, and path)`.
```